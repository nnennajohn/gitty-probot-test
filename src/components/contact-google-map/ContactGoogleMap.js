// @flow
import React from 'react';
import {
  withScriptjs,
  withGoogleMap,
  GoogleMap,
  Marker,
} from 'react-google-maps';

import markerPng from '../../images/marker.png';
import MapStyles from './map-styles.json';

/**
 *
 * @param props
 * @constructor
 * Google Map embedding component.
 */
const CMap = (props) => (
  <GoogleMap
    defaultZoom={15}
    defaultCenter={props.center}
    defaultOptions={{ styles: MapStyles }}
    mapTypeId={google.maps.MapTypeId.ROADMAP}
  >
    <Marker
      // $FlowFixMe
      position={new google.maps.LatLng(props.center.lat, props.center.lng)}
      draggable={false}
      icon={markerPng}
      title="The Playground"
      url="https://goo.gl/maps/4s6AbLht13G2"
    />
  </GoogleMap>
);

const ContactGoogleMap = withScriptjs(withGoogleMap(CMap));

export default ContactGoogleMap;
