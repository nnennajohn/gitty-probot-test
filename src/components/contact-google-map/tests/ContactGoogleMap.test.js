import React from 'react';
import ContactGoogleMap from '../index';

const testIf =
  process.env.MODE && process.env.MODE === 'skipsnapshots' ? test.skip : test;

testIf('Contact Google Map component', () => {
  const wrapper = mount(
    <ContactGoogleMap
      googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDQ6tkAoMSO8ygMYmhuA9FrOTPVFJYKhqw"
      loadingElement={<div style={{ height: `100%` }} />}
      containerElement={<div style={{ height: `400px` }} />}
      mapElement={<div style={{ height: `100%` }} />}
      center={{ lat: 40.463454, lng: -3.67186 }}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
