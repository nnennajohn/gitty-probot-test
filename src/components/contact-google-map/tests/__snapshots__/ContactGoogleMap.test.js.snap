// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Google Map component 1`] = `
<withScriptjs(withGoogleMap(CMap))
  center={
    Object {
      "lat": 40.463454,
      "lng": -3.67186,
    }
  }
  containerElement={
    <div
      style={
        Object {
          "height": "400px",
        }
      }
    />
  }
  googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyDQ6tkAoMSO8ygMYmhuA9FrOTPVFJYKhqw"
  loadingElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
  mapElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
</withScriptjs(withGoogleMap(CMap))>
`;
